### ISSUE TITLE
How to handle the negative values in bbox prediction convolution layer when calculating the IOU Loss?

### ISSUE THREAD

**Issue by Hwang64** [tianzhi0549/FCOS#25]
The outputs value of the convolution layer for bbox prediction are not always positive, so when meeting with a negative value, the IOU Loss will be 'NAN' (because the log) ,  I find that the output value of the bbox prediction are post-processed by a 'torch.exp' operation, will this operation harm the detection performance? Is there any other operations to deal with the negative values in the bbox prediction feature map?

**Comment by tianzhi0549** [comment]
@Hwang64 We do not think the `torch.exp` can harm the detection performance. In some works, they make use of `Relu` to make the regression positive. We think `exp` is better because it is differentiable everywhere.
